### ISSUE TITLE
[Lua Editor] Settings aren't correctly serialized/deserialized

### ISSUE THREAD

**Issue by yaakuro** [o3de/o3de#17840]
**Describe the bug**
Lua Editor is not "saving" properties.

**Steps to reproduce**
1. Open Lua Editor
2. Click Options/Settings menu
3. Change any value
4. Save and close dialog
5. Close and restart Lua Editor
6. Open the Settings menu again
7. Check changed property should not be changed

**Expected behavior**
The settings dialog should save and or load the changed properties correct.

**Actual behavior**
The settings are back to the default values after restarting the Lua Editor.

**Screenshots/Video**
If applicable, add screenshots and/or a video to help explain your problem.

**Found in Branch**
This issue exists already for a while.

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
a84a5cb7b8547c10ab52ced5485ff21fbc020bf3

**Desktop/Device (please complete the following information):**
 - Device:  PC
 - OS: Linux 
 - Version Ubuntu 22.04 LTS
 - CPU  Ryzen 5950x,
 - GPU AMD 6950 XT
 - Memory 128GB

**Additional context**

I was able to pinpoint the issue. For some reason this method in EditorFrameworkApplication.cpp
```cpp
    AZStd::string Application::GetApplicationGlobalStoragePath()
    {
        return QStandardPaths::writableLocation(QStandardPaths::AppDataLocation).toUtf8().data();
    }
```
does return when loading or saving different locations.

When loaded it points to: ~/.local/share/LuaIDE
When saved it is using: ~/.local/share/O3DE/O3DE Editor/LuaIDE

Didn't fully dig into the why. Nevertheless I would have a suggestion. Wouldn't it be better to have the properties saved in the user folder of the project? Using the user folder would allow to have different properties for the Lua Editor per project.
I do this:

```cpp
    AZStd::string Application::GetApplicationGlobalStoragePath()
    {
        return AZ::Utils::GetProjectUserPath().c_str();
    }
```

This puts the files into:

O3deProject/user/LuaIDE/

And everything works.

**Comment by yaakuro** [comment]
New update on this. It looks like, that the de-serialization using those paths happens before:

UIFramework.cpp:Framework::Run is called.

In this function however the following lines are called:

        pApplication->setOrganizationName("O3DE");
        pApplication->setApplicationName("O3DE Editor");

This changes the application name meaning it changes the location of

QStandardPaths::writableLocation(QStandardPaths::AppDataLocation)

**Comment by derikson-dev** [comment]
Very annoying to load the Lua Editor and reconfigure the settings every time. Can this issue be resolve in the next release?

**Comment by yaakuro** [comment]
> Very annoying to load the Lua Editor and reconfigure the settings every time. Can this issue be resolve in the next release?

You still have that issue? This should have fixed it: https://github.com/o3de/o3de/pull/17841

**Comment by derikson-dev** [comment]
> > Very annoying to load the Lua Editor and reconfigure the settings every time. Can this issue be resolve in the next release?
> 
> You still have that issue? This should have fixed it: #17841

Yes. Problem persist in the Lua Editor.

**Comment by yaakuro** [comment]
> > > Very annoying to load the Lua Editor and reconfigure the settings every time. Can this issue be resolve in the next release?
> > 
> > 
> > You still have that issue? This should have fixed it: #17841
> 
> Yes. Problem persist in the Lua Editor.

Make sure you are on the development branch.  Also try to change a setting and click, "Save & Close". Then have a look in
MyProject/user/LuaIDE 
In that folder you should have GlobalUserSettings.xml, check if the value changed. If you don't see that file, then there is another issue.
